src/main/java/bookstore/model/ShoppingCart.java

```diff
@@ -0,0 +1,37 @@
+package bookstore.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.MapsId;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.PrimaryKeyJoinColumn;
+import jakarta.persistence.Table;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE shopping_carts SET is_deleted = TRUE WHERE id = ?")
+@Where(clause = "is_deleted = FALSE")
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    //@GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne(cascade = CascadeType.ALL)
+    @MapsId
+    @PrimaryKeyJoinColumn(name = "user_id")
+    private User user;
+    @OneToMany(mappedBy = "shoppingCart")
+    private Set<CartItem> cartItems = new HashSet<>();
+    @Column(name = "is_deleted", nullable = false)
+    private Boolean isDeleted = false;
```

Comment:
I think the `isDeleted` field is redundant here.
 
For example, if the user has been deleted, so our shopping cart has a reference to a deleted user that becomes inactive (so, no need to soft delete the shopping cart)